packages/upload-core/src/upload.js

```diff
@@ -267,6 +267,20 @@ class Upload {
         return false;
       }
     }
+    if (this.options.allowedFileNameCharacters) {
+      const fileName = this.file.name.substring(
+        0,
+        this.file.name.lastIndexOf('.')
+      );
+      const regExp = new RegExp(
+        `([^${this.options.allowedFileNameCharacters}])`,
+        'g'
+      );
+      if (fileName && fileName.match(regExp) !== null) {
```

Comment:
I would use `regExp.test(fileName)` which is already safe if `fileName` is `undefined`. It also returns a boolean so there no need to explicitly check for `null`.
You would probably have to invert the logic though.